examples/unimp_arxiv.py

```diff
@@ -0,0 +1,170 @@
+import logging
+import os
+
+import torch
+from ogb.nodeproppred import PygNodePropPredDataset
+
+import torch_geometric.transforms as T
+import torch_geometric.utils.mask as mask_util
+from torch_geometric.nn import TransformerConv
+
+logger = logging.getLogger(__file__)
+logging.basicConfig(level=logging.INFO)
+
+# Get dataset
+root = os.path.join("..", "data", "OGB")
+dataset = PygNodePropPredDataset(
+    "ogbn-arxiv",
+    root,
+    transform=T.Compose([T.ToUndirected()]),
+)
+data = dataset[0]
+num_classes = 40
+
+# Form masks for labels: mask gives labels being predicted
+split_idx = dataset.get_idx_split()
+train_mask = mask_util.index_to_mask(split_idx["train"], size=data.num_nodes)
+valid_mask = mask_util.index_to_mask(split_idx["valid"], size=data.num_nodes)
+test_mask = mask_util.index_to_mask(split_idx["test"], size=data.num_nodes)
+
+# Model parameters
+inner_dim = 16
+heads = 2
+
+
+def ratio_mask(mask: torch.Tensor, ratio: float, shuffle: bool = False):
+    r"""Modifies the existing :obj:`mask` by additioning setting :obj:`ratio`
+    of the :obj:`True` entries to :obj:`False`. Does not operate inplace.
+
+    If shuffle is required the masking proportion is not exact.
+
+    Args:
+        mask (torch.Tensor): The mask to re-mask.
+        ratio (float): The ratio of entries to remove.
+        shuffle (bool): Whether or not the mask is pre-shuffled, if so there is
+            no need to randomize which entires are set to :obj:`False`.
+    """
+    n = int(mask.count_nonzero().item())
+    new_mask = torch.ones(len(mask), dtype=torch.bool)
```

Comment:
can't we initialize `new_mask` based on `mask`? Since we are doing `mask & new_mask` later anyway.

Reply:
good point - do you think clone is the cleanest way to do this?